packages/textkit/src/engines/linebreaker/index.js

```diff
@@ -17,6 +18,48 @@ const opts = {
   shrink: 9,
 };
 
+/**
+ * Get regions of urls in the attributed string
+ *
+ * @param {Object} attributedString attributed string
+ * @returns {Array} regions of urls
+ */
+const getUrlRegions = attributedString => {
+  const urlRegions = [];
+
+  for (let index = 0; index < attributedString.runs.length; index += 1) {
+    const { start, end } = attributedString.runs[index];
+    const strPart = attributedString.string.slice(start, end);
+    if (isUrl(strPart)) {
+      urlRegions.push({ start, end });
+    }
+  }
+
+  return urlRegions;
+};
+
+/**
+ * Check whether the line is in the URL region
+ *
+ * @param {Object[]} urlRegions URL Regions
+ * @param {Number} attributedStringLineStartPosition attributed string line start position
+ * @returns {Boolean} if the line is in the URL region return true otherwise false
+ */
+const isInUrlRegion = (urlRegions, attributedStringLineStartPosition) => {
```

Comment:
Why is the line start position used here? 
Are we not supposed to determine if the last character of a line belongs to a URL, and if it does, avoid adding a hyphen?

Reply:
fixed